File: second-year/second-semester/MPP/examen/rezolvari/examen_pb2/ExamenNetworking/src/main/java/exam/networking/rpcprotocol/Request.java
package exam.networking.rpcprotocol;

import java.io.Serializable;

public class Request implements Serializable {
    private RequestType type;
    private Object data;

    private Request() {
    }

    public RequestType getType() {
        return type;
    }

    public Object getData() {
        return data;
    }

    @Override
    public String toString() {
        return "Request{" +
                "type='" + type + '\'' +
                ", setData='" + data + '\'' +
                '}';
    }
    private void setData(Object data) {
        this.data = data;
    }

    private void setType(RequestType type) {
        this.type = type;
    }


    public static class Builder {
        private Request request = new Request();

        public Builder setType(RequestType type) {
            request.setType(type);
            return this;
        }

        public Builder setData(Object data) {
            request.setData(data);
            return this;
        }

        public Request build() {
            return request;
        }
    }



}

File: second-year/second-semester/MPP/mpp-proiect-repository-raulbaciulescu/travelAgency2/networking/Request.cs
namespace networking
{    
    [Serializable]
    public class Request
    {
        public RequestType type { get; set; }
        public Object data { get; set; }

        public override string ToString()
        {
            return "Request{" +
                   "type='" + type + '\'' +
                   ", data='" + data + '\'' +
                   '}';
        }
        public class Builder{
            private static Request request = new Request();

            public Builder Type(RequestType type) {
                request.type = type;
                return this;
            }

            public Builder Data(Object data) {
                request.data = data;
                return this;
            }

            public Request Build() {
                return request;
            }
        }
    }    
}

